homeassistant/components/blebox/button.py

```diff
@@ -15,20 +17,23 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up a BleBox button entry."""
-    create_blebox_entities(
-        hass, config_entry, async_add_entities, BleBoxButtonEntity, "buttons"
-    )
+    entities: list[BleBoxButtonEntity] = []
+
+    for feature in get_blebox_features(hass, config_entry, "buttons"):
+        entities.append(BleBoxButtonEntity(feature))
```

Comment:
The typing isn't completely correct unfortunately. That is the pyright warning
```
Argument of type "Feature" cannot be assigned to parameter "feature" of type "Switch" in function "__init__"
  "Feature" is incompatible with "Switch"
```

To fix it, you'll probably need to use a few overloads for `get_blebox_features`. Like so
```py
@overload
@callback
def get_blebox_features(
    hass: HomeAssistant, config_entry: ConfigEntry, entity_type: Literal["buttons"]
) -> Sequence[blebox_uniapi.button.Button]:
    ...

@overload
@callback
def get_blebox_features(
    hass: HomeAssistant, config_entry: ConfigEntry, entity_type: Literal["climates"]
) -> Sequence[blebox_uniapi.climate.Climate]:
    ...

# ...

@callback
def get_blebox_features(
    hass: HomeAssistant, config_entry: ConfigEntry, entity_type: str
) -> Sequence[Feature]:
    # ... Implementation ...
```

Also note the use of `Sequence` instead of `list`. That will make the overloads a bit easier.

Reply:
I figured that adding the overloads didn't really make sense inside `__init__.py`
The idea is to make the code simpler - not more complicated.

Instead I adjusted each platform, and I got rid of `get_blebox_features` altogether.
I think that works...